src/modules/local_position_estimator/BlockLocalPositionEstimator.cpp

```diff
@@ -390,15 +392,35 @@ void BlockLocalPositionEstimator::update()
 		mavlink_and_console_log_info(&mavlink_log_pub, "[lpe] reinit x");
 	}
 
-	// reinitialize P if necessary
+	// force P symmetry and reinitialize P if necessary
 	bool reinit_P = false;
 
 	for (int i = 0; i < n_x; i++) {
-		for (int j = 0; j < n_x; j++) {
-			if (!PX4_ISFINITE(_P(i, j))) {
+		for (int j = 0; j <= i; j++) {
+			if (_P(i, j) > P_MAX) {
+				// if we aren't measuring terrain state etc.
+				// then it will grow to this max, don't
+				// want to reset here or it will be a
+				// periodic reset
+				_P(i, j) = P_MAX;
```

Comment:
@jgoppert Based on experience with other estimators definitely a consistent constraint / reset is important.